client.go

```diff
@@ -357,3 +367,153 @@ func prepareURL(url *url.URL, path string) (*url.URL, error) {
 
 	return u, nil
 }
+
+// prepareQueryParameters Allows for easy preparation of query string
+func (cl *Client) prepareQueryParameters(params interface{}) (queryString string, err error) {
+	v, err := query.Values(params)
+
+	// If we do have account switch key - we will add it and toggle ASK back to disabled
+	if cl.accountSwitchEnabled == true {
+		v.Add("accountSwitchKey", cl.accountSwitchKey)
+		cl.accountSwitchEnabled = false
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	return v.Encode(), nil
+}
+
+// makeAPIRequest creates an HTTP request that can be sent to Akamai APIs. It will handle security headers and signinig of the request.
+//
+func (cl *Client) makeAPIRequest(method, path string, queryParams, structResponse, structRequest interface{}, headers map[string]string) (*ClientResponse, error) {
+
+	log.Debug("[NewRequest]::Prepare URL for http request")
+	targetURL, _ := prepareURL(cl.baseURL, path)
+
+	log.Debug("[NewRequest]::Create http request")
+	req, err := http.NewRequest(method, targetURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	/*
+		Modify request for POST/PUT
+	*/
+	if method == http.MethodPost || method == http.MethodPut {
+		log.Info("Prepare request body object")
+		log.Debug("[NewRequest]::Method is POST/PUT")
```

Comment:
Why not specify exact method instead of PUT/POST guess

Reply:
https://github.com/apiheat/go-edgegrid/pull/21/commits/9dac2f34486a2acc758701b882162d3aa6196871